chore: Update docs for building Zebra

upbqdn opened the pull request:
## Motivation

Address https://github.com/ZcashFoundation/zebra/issues/7993#issuecomment-1959163713 before #7993.

### PR Author Checklist

#### Check before marking the PR as ready for review:
  - [x] Will the PR name make sense to users?
  - [x] Does the PR have a priority label?
  - [x] Have you added or updated tests?
  - [x] Is the documentation up to date?

## Solution

- Refer to the upcoming Zebra 1.6 instead of 1.5 for shielded scanning.
- Mention that building Zebra now requires `protoc` no older than v3.12.

### Reviewer Checklist

Check before approving the PR:
  - [ ] Does the PR scope match the ticket?
  - [ ] Are there enough tests to make sure it works? Do the tests cover the PR motivation?
  - [ ] Are all the PR blockers dealt with?
        PR blockers can be dealt with in new tickets or PRs.

_And check the PR Author checklist is complete._

upbqdn committed:
Update build instructions with scanning enabled

oxarbitrage commented on book/src/user/shielded-scan.md:
````diff
@@ -24,6 +24,12 @@ install the latest release:
 cargo install --features shielded-scan --locked --git https://github.com/ZcashFoundation/zebra zebrad
 ```
 
+> [!NOTE] Enabling the `shielded-scan` feature will make Zebra use the
````
Unfortunately this not only happen when you have the shielded-scan feature enabled but when building zebra without any flag too.

upbqdn replied:
Fixed in [7594ccf](https://github.com/ZcashFoundation/zebra/pull/8315/commits/7594ccfbb2a7a3c0dae1598ad28f43404827d8e1).

upbqdn committed:
Remove `pkg-config` from build deps in README

I tried compiling Zebra with `--all-features` on a Linux machine without
`pkg-config` or `pkfgconf` (which is an alternative to `pkg-config`),
and it worked.

upbqdn committed:
Add `protoc` to build deps in README

arya2 reviewed (approved):
Thank you!
